Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/CheckInMatching.kt

```diff
@@ -0,0 +1,75 @@
+package de.rki.coronawarnapp.presencetracing.risk
+
+import de.rki.coronawarnapp.eventregistration.checkins.CheckIn
+import de.rki.coronawarnapp.eventregistration.checkins.download.TraceTimeIntervalWarningPackage
+import de.rki.coronawarnapp.server.protocols.internal.pt.TraceWarning
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.toLocalDate
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import kotlinx.coroutines.withContext
+import org.joda.time.Duration
+import org.joda.time.Instant
+import kotlin.coroutines.CoroutineContext
+
+suspend fun launchMatching(
+    checkIns: List<CheckIn>,
+    warningPackages: List<TraceTimeIntervalWarningPackage>,
+    coroutineContext: CoroutineContext
+): Collection<Deferred<List<CheckInWarningOverlap>>> {
+    val launcher: CoroutineScope.(
+        List<CheckIn>,
+        TraceTimeIntervalWarningPackage
+    ) -> Deferred<List<CheckInWarningOverlap>> =
+        { list, warningPackage ->
+            async {
+                findMatches(list, warningPackage)
+            }
+        }
+
+    return warningPackages.map { warningPackage ->
```

Comment:
Please add a unit test that shows how this is expected to behave if there is an error during execution.

Reply:
Most likely, it would be a file reading issue. No idea what should happen there. @mlenkeit What should happen if a package cannot be unpacked/evaluated? Omit the package or fail the whole calculation?

Reply:
Abort the calculation, ideally show a white card on the home screen.

Reply:
I will fix it in the other PR